packages/forklift-console-plugin/src/modules/Plans/views/details/tabs/VirtualMachines/modals/PlanVMsEditModal.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import { PlanEditAction } from 'src/modules/Plans/utils/types/PlanEditAction';
+import { PlanEditPage } from 'src/modules/Plans/views/edit/PlanEditPage';
+import { useModal } from 'src/modules/Providers/modals';
+import { useInventoryVms } from 'src/modules/Providers/views';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import {
+  NetworkMapModelGroupVersionKind,
+  ProviderModelGroupVersionKind,
+  StorageMapModelGroupVersionKind,
+  V1beta1NetworkMap,
+  V1beta1Plan,
+  V1beta1Provider,
+  V1beta1StorageMap,
+} from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Modal, ModalVariant } from '@patternfly/react-core';
+
+import './PlanVMsDeleteModal.style.css';
+import './PlanVMsEditModal.style.css';
+
+export interface PlanVMsEditModalProps {
+  plan: V1beta1Plan;
+  editAction: PlanEditAction;
+}
+
+export const PlanVMsEditModal: React.FC<PlanVMsEditModalProps> = ({ plan, editAction }) => {
+  const { toggleModal } = useModal();
+  const { t } = useForkliftTranslation();
+  const projectName = plan?.metadata?.namespace;
+
+  // Retrieve k8s source provider
+  const [sourceProvider, sourceProviderLoaded, sourceProviderLoadError] =
+    useK8sWatchResource<V1beta1Provider>({
+      groupVersionKind: ProviderModelGroupVersionKind,
+      namespaced: true,
+      name: plan?.spec?.provider?.source?.name,
+      namespace: plan?.spec?.provider?.source?.namespace,
```

Comment:
Please add selectors in a shared place and use them across this PR when needed. We will consume it in other pr's in the future as well.